ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/validation/StatementValidator.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License; you may not use this file
+ * except in compliance with the License.  You may obtain a copy of the License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.rest.server.validation;
+
+import io.confluent.ksql.KsqlExecutionContext;
+import io.confluent.ksql.parser.KsqlParser.PreparedStatement;
+import io.confluent.ksql.util.KsqlConfig;
+import io.confluent.ksql.util.KsqlException;
+import java.util.Map;
+
+/**
+ * An interface that allows for arbitrary validation code of a prepared statement
+ * against a point-in-time execution engine.
+ */
+@FunctionalInterface
+public interface StatementValidator {
```

Comment:
As mentioned above, I'd share the same interface for validators and executors, as there as some validation steps that just involve executing the custom executor to see if it throws.

The only differences between the two are:
- executor returns something, (but the validation can just ignore the return value)
- executor takes a ServiceContext, (though I think this will go away with my changes in #2431).

Reply:
The other difference is `KsqlEngine` vs `KsqlExecutionContext`. my newest patch resolves this difference by adding `getPersistentQueries` to the execution context

Reply:
ServiceContext does not go away - `ListTopics` requires a topic client.